dotcom-rendering/src/web/server/index.ts

```diff
@@ -184,6 +187,36 @@ export const renderKeyEvents = (
 	}
 };
 
+/**
+ * Sends markup to Frontend, which forwards it to the client.
+ * Frontend expects that it will receive the markup string as a value of the
+ * key 'html' in a json object, i.e. { "html": "<some markup>" }
+ */
+export const renderCards = (
+	{ body }: { body: CardsRequest },
+	res: express.Response,
+): void => {
+	try {
+		const { cards, startIndex, config } = body;
+
+		const dcrContainerPalette = decideContainerPalette(
+			config.metadata?.map((meta) => meta.type),
+		);
+		const dcrTrails = enhanceCards(cards, dcrContainerPalette, startIndex);
+
+		const html = cardsToHtml({
+			cards: dcrTrails,
+			containerPalette: dcrContainerPalette,
+		});
+
+		res.status(200).send({ html });
+	} catch (e) {
+		// todo: need to replicate error-handling behaviour of existing component
```

Comment:
I can't really see anything useful that Frontend could pass to us here if something breaks down either within Frontend or within DCR while it's trying to fetch the new cards. Given this, I'm tempted to hard-code a html string in the ShowMore.Importable component which contains a generic error message to put into the DOM if it receives anything other than a 200 response which has a `html` key in the response body. Does anyone have thoughts on this?

Reply:
This is an interesting question!

I think a generic error message would be acceptable. It would be interesting to share with designers and UX to get their thoughts.

Reply:
thanks! I've double-checked the contract with Frontend with James, and have added a hard-coded message on the client side which will be displayed if it receives an error from Frontend. This (more-or-less) replicates the existing Frontend-rendered error-handling behaviour, but I agree that it would be good to revisit the UX and design of this component with others. I'd like to leave this for another PR to avoid more scope creep, or do you think that this is a blocker for this PR?